libs/pika/threading_base/include/pika/threading_base/thread_data.hpp

```diff
@@ -433,6 +433,24 @@ namespace pika::threads::detail {
         }
 #endif
 
+        // convenience object to temporarily change priority
+        struct scoped_thread_priority
+        {
+            pika::execution::thread_priority old_priority_;
+            scoped_thread_priority(pika::execution::thread_priority new_p)
+              : old_priority_{
+                    threads::detail::get_self_id_data()->get_priority()}
+            {
+                threads::detail::get_self_id_data()->set_priority(new_p);
+            }
```

Comment:
```suggestion
            explicit scoped_thread_priority(pika::execution::thread_priority new_priority)
              : old_priority_{
                    threads::detail::get_self_id_data()->get_priority()}
            {
                threads::detail::get_self_id_data()->set_priority(new_priority);
            }
```

Reply:
The `explicit` would still be good to have here (I don't care much about the parameter name). This could also be made immovable.